### ISSUE TITLE
PossiblyUnusedParam error reported despite parameter usage

### ISSUE THREAD

**Issue by grizzdrasil** [weirdan/doctrine-psalm-plugin#159]
Hi there !

I'm currently using Psalm v6.0.0 (Problem already happening in Psalm v5.2) configured at `errorLevel` 3, with the `psalm/plugin-symfony` and `weirdan/doctrine-psalm-plugin` plugins installed.

Psalm reports several PossiblyUnusedParam as `ERRORS` despite these parameters being used (seemingly can't be unused).

The documentation recommends prefixing such parameters with `_` to ignore this error, which I tried but to no avail.

It resembles a clear false positive to me given the following example :

>ERROR: PossiblyUnusedParam
at /home/project/src/Repository/ApiTokenRepository.php:12:49
Param #1 is never referenced in this method (see https://psalm.dev/134)
    public function __construct(ManagerRegistry $registry)

>ERROR: PossiblyUnusedParam
at /home/project/src/Repository/ApiTokenRepository.php:17:40
Param #1 is never referenced in this method (see https://psalm.dev/134)
    public function getApiToken(string $token): ?ApiToken

```php
namespace App\Repository;

use App\Entity\ApiToken;
use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
use Doctrine\ORM\AbstractQuery;
use Doctrine\Persistence\ManagerRegistry;

/**
 * @extends ServiceEntityRepository<ApiToken>
 */
class ApiTokenRepository extends ServiceEntityRepository
{
    public function __construct(ManagerRegistry $registry)
    {
        parent::__construct($registry, ApiToken::class);
    }

    public function getApiToken(string $token): ?ApiToken
    {
        $qb = $this->createQueryBuilder('apiToken')
                   ->andWhere('apiToken.token = :token')
                   ->setParameter('token', $token);

        /** @var ?ApiToken $result */
        $result = $qb->getQuery()
                     ->getOneOrNullResult(AbstractQuery::HYDRATE_OBJECT);

        return $result;
    }
}
```

Thank you for your insight !

**Comment by ThomasLandauer** [comment]
Same as https://github.com/psalm/psalm-plugin-doctrine/issues/157 ?

**Comment by grizzdrasil** [comment]
> Same as [#157](https://github.com/psalm/psalm-plugin-doctrine/issues/157) ?

Not exactly, this is a `PossiblyUnusedParam` error despite it being clearly used anytime in the method.

It is indeed close, maybe the same detection method that triggers both errors though.

**Comment by YoYoShik** [comment]
Hi! I got the same error. Every parameter of every method of every repository.
Did you find any solution? (Apart from the obvious ignoring PossiblyUnusedParam in the repositories?:) )
@grizzdrasil

**Comment by mykolastepanenko** [comment]
The same error
